src/main/java/Butler.java

```diff
@@ -0,0 +1,216 @@
+                    try {
+                        summary = inputList[1].split(" /by ", 2)[0];
+                        String deadline = inputList[1].split(" /by ", 2)[1];
+                        LocalDate formattedDeadline = LocalDate.parse(deadline);
+                        addedTask = new DeadlineTask(summary, formattedDeadline);
+                        taskList.add(addedTask);
+                        reply += "\nI have added your Deadline task as follows:\n"
+                                + "Added: " + addedTask + "\n";
+                        isValidCommand = true;
+                    } catch (DateTimeParseException e) {
+                        System.out.println("\nPlease input a valid Date format.\n");
+                    } catch (Exception e) {
+                        System.out.println("\nExcuse me, " +
+                                "please provide a summary and deadline using the tag /by.\n");
+                    }
+                    break;
+
+                case "event":
+                    try {
+                        summary = inputList[1].split(" /at ", 2)[0];
+                        String time = inputList[1].split(" /at ", 2)[1];
+                        LocalDate startDate = LocalDate.parse(time.split(" ")[0]);
+                        LocalDate endDate = LocalDate.parse(time.split(" ")[1]);
+                        addedTask = new EventTask(summary, startDate, endDate);
+                        taskList.add(addedTask);
+                        reply += "\nI have added your Event task as follows:\n"
+                                + "Added: " + addedTask + "\n";
+                        isValidCommand = true;
+                    } catch (DateTimeParseException e) {
+                        System.out.println("\nPlease input a valid Date format.\n");
+                    } catch (Exception e) {
+                        System.out.println("\nExcuse me, " +
+                                "please provide a summary and time of event using the tag /at.\n");
+                    }
+                    break;
+
+                default:
+                    reply += "\nI am sorry, but I do not understand this command.\n";
+
+            }
+
+            if (isNew && isValidCommand) {
+                System.out.println(reply);
+                writeToFile(input);
+            }
+        } catch (Exception e) {
+            System.out.println("\nInvalid command given.\n");
+        }
+    }
+
+    private static void deleteTask(String input, ArrayList<Task> taskList, Boolean isNew) {
+        String reply;
+        try {
+            int index = Integer.parseInt(input.split(" ")[1]);
+            taskList.remove(index-1);
+            if (isNew) {
+                reply = "\nTask " + index + " has been deleted.\n";
+                System.out.println(reply);
+                writeToFile(input);
+            }
+        } catch (Exception e) {
+            System.out.println("\nPlease provide a valid index.\n");
```

Comment:
Can we assume that all thrown exceptions are only due to invalid index (and not other unchecked exceptions)?

Reply:
Thank you for the feedback. I have refactored my code such that exceptions caught are specific instead of blanket catching. This will allow me to be aware of other exceptions that may be thrown as well.